library/src/main/java/com/datatorrent/lib/util/PojoUtils.java

```diff
@@ -696,11 +662,97 @@ private static Object createSetter(Class<?> pojoClass, String setterExpr, String
+   */
+  public static Expression createExpression(Class<?> inputType, String expr, Class<?> returnType)
+  {
+    return createExpression(inputType, expr, returnType, null);
+  }
+
+  /**
+   * This method takes in expression, compiles the expression to provide a executable form of expression.
+   * This methods also takes in list of classes and method which can be imported statically in expression.
+   * <p/>
+   * This method uses {@link JavaExpressionParser} as expression parser.
+   *
+   * @param inputType      Type of input object
+   * @param expr           expression to be compiled.
+   * @param returnType     Return type of the expression.
+   * @param defaultImports List of classes/method which will be statically imported to expression compilation.
+   * @return Object of type {@link Expression} which can be directly executed.
+   */
+  public static Expression createExpression(Class<?> inputType, String expr, Class<?> returnType,
+      String[] defaultImports)
+  {
+    JavaExpressionParser javaExpressionParser = new JavaExpressionParser();
+    javaExpressionParser.setInputObjectPlaceholder("$", PojoUtils.OBJECT);
+
+    return createExpression(inputType, expr, returnType, defaultImports, javaExpressionParser);
+  }
+
+  /**
+   * This method takes in expression, compiles the expression to provide a executable form of expression.
+   * This methods also takes in list of classes and method which can be imported statically in expression.
+   * <p/>
+   * Using this method one can override expression parser implementation.
+   *
+   * @param inputType      Type of input object
+   * @param expr           expression to be compiled.
+   * @param returnType     Return type of the expression.
+   * @param defaultImports List of classes/method which will be statically imported to expression compilation.
+   * @param parser         Expression parser that should be used to parse expression.
+   * @return Object of type {@link Expression} which can be directly executed.
+   * @see {@link JavaExpressionParser} as a example.
+   */
+  public static Expression createExpression(Class<?> inputType, String expr, Class<?> returnType,
+      String[] defaultImports, Expression.ExpressionParser parser)
+  {
+    String code = parser.convertToCompilableExpression(expr, inputType, returnType);
+
+    return (Expression)compileExpression(code, Expression.class, new String[] {PojoUtils.OBJECT}, defaultImports);
+  }
+
+  private static Object compileExpression(String code, Class<?> implClass, String[] params)
+  {
+    return compileExpression(code, implClass, params, null);
+  }
+
+  private static Object compileExpression(String code, Class<?> implClass, String[] params, String[] defaultImports)
+  {
+    List<String> imports = new LinkedList<>();
+    if (defaultImports != null && defaultImports.length != 0) {
+      for (String defaultImport : defaultImports) {
+        if (!defaultImport.startsWith("static")) {
```

Comment:
When adding a given import as static, that means the method will be directly available in expression to be used. This way, the expression writer can use similar syntax like excel functions.